bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/InterleavedLedgerStorage.java

```diff
@@ -191,6 +191,11 @@ public void diskJustWritable(File disk) {
         };
     }
 
+    @Override
+    public void forceGC() {
+        gcThread.enableForceGC();
```

Comment:
this is not actually implementing a "trigger" behavior. it only sets flag, however the actual gc will be happening after `gcWaitInterval`. so I think a better solution would be:

```suggestion
        gcThread.enableForceGC();
        // submit a task immediately
        gcExecutor.submit(this);
```

Reply:
Thanks, Maybe the original name is misleading. Here is what enableForceGC does in gcThread:
```
    public void enableForceGC() {
        if (forceGarbageCollection.compareAndSet(false, true)) {
            LOG.info("Forced garbage collection triggered by thread: {}", Thread.currentThread().getName());
            triggerGC(true, suspendMajorCompaction.get(),
                      suspendMinorCompaction.get());
        }
    }

    Future<?> triggerGC(final boolean force,
                        final boolean suspendMajor,
                        final boolean suspendMinor) {
        return gcExecutor.submit(() -> {
                runWithFlags(force, suspendMajor, suspendMinor);
            });
    }

```

Reply:
i see